### PR TITLE
Add Config partition support.

### PR DESCRIPTION
**Pull request by gianlzed**
Required changes:

1. Change make-flash to use config.img to create config partition.
2. Change Makefile to pass config.img to make-flash
3. Change test-cert to become the container that creates config.img
4. Remove test-cert from zededa container and mount partition at start.

Note: In a perfect world we would use LinuxKit's own pkg/mount, we can't
because the FAT filesystem (nor squashfs) is not supported, and can
work with UUID/Lables only on ext4, btrfs, xfs.

### CODE REVIEW

File: maketestconfig.sh [hunk 1]
```diff
@@ -0,0 +1,23 @@
+#!/bin/sh
+# Usage:
+#
+#      ./makeflash.sh <output.img>
+#
+MKCONFIG_TAG="$(linuxkit pkg show-tag pkg/test-cert)-amd64"
+
+# Make sure it exists
```
Comment (eriknordmark) [inline maketestconfig.sh]: I don't see a check for existence

File: pkg/test-cert/make-config [hunk 2]
```diff
@@ -0,0 +1,17 @@
+#!/bin/sh
+
+set -ex
+
+CONFIG_FILE=$1
+# Config is 1Mb
+CONFIG_SIZE_KB=4096
```
Comment (eriknordmark) [inline pkg/test-cert/make-config]: 1Mb or 4096Kb?

File: maketestconfig.sh (hunk 1 above)
Comment (gianlzed) [reply maketestconfig.sh]: Yes, artifact from previous iterations. Creating it here showed problems on Mac.

File: pkg/test-cert/make-config (hunk 2 above)
Comment (gianlzed) [reply pkg/test-cert/make-config]: Oh right, yet another artifact with the "let's try ext4"

**Comment by gianlzed** [comment]
@eriknordmark I had no idea ztools did that!

I will copy in test-cert --from=ztools /config /conf but it will be definitely confusing because then on the rootfs of the zededa container we will still have a falsely populated /config that we will mount on.

I don't think an overlayfs makes sense on a config partition either.

**Comment by rvs** [comment]
Approach looks OK in general, but before we merge this, I'd like to raise a few issues:
  1. I think re-using test-cert container for this configuration creation is confusing. I'd rather create a separate container called conf that holds the configuration files AND the logic for the export (al the msdos FS creation tools, etc.)
  2. This dedicated conf container will then inherit (via multiple FROM's) from any other container from which it needs to pull the configuration files into a centralized location. This will obviously include ztools and test-cert

I think this will provide a much cleaner separation between conf and everything else that goes in there.

**Comment by gianlzed** [comment]
This is exactly how I am changing test-cert, although it's not too visible here, it gets more visible in the upcoming patch, which btw renames test-cert as test-conf.

Essentially the make-config script, added in the test-cert (read: test-conf) is what makes it a "configuration generator" container.  Its Dockerfile tells where to get the data from. Right now it takes it from itself (the test-certs, being the test conf). It will now also take them from the ztools, as there's a /config partition there, as pointed out by erik.

Essentially with this scheme in the future we could create any configuration parititon generator by copying make-config and setting it as the entry point, and defining in the Dockerfile where to pull things from.

Makes sense?

I might separate this test-conf container from the test-cert container, that only contains the certs, but it looks like a unneeded duplication.

**Comment by gianlzed** [comment]
forced pushed new patch to the same branch (https://github.com/zededa/zenbuild/pull/20/commits/77e45a6c106f69a5fea5e4fa6f10fe733c39d9d2), writing here as github doesn't make it that visible.

**Comment by gianlzed** [comment]
ping?

**Pull request merged**